### PR TITLE
Fix unnecessary trimming of line resulting in incorrect indexi…

### LINKED ISSUE #11669: Still some bugs in error formatting, indent somehow involved
# Steps to reproduce

**N.B.** The repro code is indented. **THIS IS CRUCIAL TO THE REPRO.** Include the indent when copying and pasting into your console window.
```powershell
        $CopyErrors = New-Object 'System.Collections.ArrayList'

        function CopyFiles( $srcRoot, $destRoot, $plat )
        {
            Write-Host "Copying binaries from $($srcRoot)..." -Fore Green

            $fullSrc = 'C:\Windows\system32\ntdll.dll'
            $fullDst = 'C:\does\not\exist'

            Copy-Item $fullSrc $fullDst -ErrorVariable +CopyErrors
        }

        CopyFiles

        $CopyErrors

        $error.Count

        $error

        $error.Count
```

# Expected behavior

You should see 1 error, from outputting `$CopyErrors`, then `$error.Count` should output "1", then `$error` should show that same error again, then `$error.Count` should still be "1".

# Actual behavior

During the formatting of `$CopyErrors`, things go sideways, and the errors don't appear in the console, but carnage starts accumulating in `$error`. Here are the printable errors:

```
MethodInvocationException:
Line |
 122 |  …             $line = $line.Insert($offsetInLine

     |                ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
     | Exception calling "Insert" with "2" argument(s): "Specified argument was out of the range of valid values. (Parameter 'startIndex')"

MethodInvocationException:
Line |
 122 |  …             $line = $line.Insert($offsetInLine

     |                ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
     | Exception calling "Insert" with "2" argument(s): "Specified argument was out of the range of valid values. (Parameter 'startIndex')"
```

# Environment data

```none
Name                           Value
----                           -----
PSVersion                      7.0.0-rc.2
PSEdition                      Core
GitCommitId                    7.0.0-rc.2
OS                             Microsoft Windows 10.0.18363
Platform                       Win32NT
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0…}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0
```

### PR DESCRIPTION
**Pull request by SteveL-MSFT**
# PR Summary

After the switch to use `InvocationInfo.PositionMessage` instead of crafting equivalent, the use of `Trim()` caused a mismatch between the line of script with the error and the underlining line so the line of code to insert VT100 color can be out of bounds.  The whitespace at the beginning of the line causes this issue because `Trim()` makes the highlight line too short (previously the script line would also be trimmed).

Also fixed an issue where it was splitting `PositionMessage` by `+` to get the individual parts, but the message itself can contain a `+` so using newline instead.  This with the trim is what caused the index to be out of bounds as the line is now too short to insert the VT100 code to reset the color back.

In addition, switched all use of `` `n`` to `[Environment]::Newline`.

Finally, removed an extra newline in the rendering.

## PR Context

Fix https://github.com/PowerShell/PowerShell/issues/11669

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
    - [x] None
    - **OR**
    - [ ] [Experimental feature(s) needed](https://github.com/MicrosoftDocs/PowerShell-Docs/blob/staging/reference/6/Microsoft.PowerShell.Core/About/about_Experimental_Features.md)
        - [ ] Experimental feature name(s): 
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - [ ] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed: 
- **Testing - New and feature**
    - [ ] N/A or can only be tested interactively
    - **OR**
    - [x] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
- **Tooling**
    - [x] I have considered the user experience from a tooling perspective and don't believe tooling will be impacted.
    - **OR**
    - [ ] I have considered the user experience from a tooling perspective and enumerated concerns in the summary. This may include:
        - Impact on [PowerShell Editor Services](https://github.com/PowerShell/PowerShellEditorServices) which is used in the [PowerShell extension](https://github.com/PowerShell/vscode-powershell) for VSCode (which runs in a different PS Host).
        - Impact on Completions (both in the console and in editors) - one of PowerShell's most powerful features.
        - Impact on [PSScriptAnalyzer](https://github.com/PowerShell/PSScriptAnalyzer) (which provides linting & formatting in the editor extensions).
        - Impact on [EditorSyntax](https://github.com/PowerShell/EditorSyntax) (which provides syntax highlighting with in VSCode, GitHub, and many other editors).

### CODE REVIEW

File: src/System.Management.Automation/FormatAndOutput/DefaultFormatters/PowerShellCore_format_ps1xml.cs [hunk 1]
```diff
@@ -1100,18 +1100,18 @@ private static IEnumerable<FormatViewDefinition> ViewsOf_System_Management_Autom
                                                 $offsetInLine = 0
```
Comment (PoshChan) [inline src/System.Management.Automation/FormatAndOutput/DefaultFormatters/PowerShellCore_format_ps1xml.cs]: @SteveL-MSFT, your last commit had 2 failures in `PowerShell-CI-linux`
ConsoleHost unit tests.Pipe to/from powershell.errors are in text if error is redirected, encoded command, non-interactive, and outputformat specified
```powershell
Expected strings to be the same, but they were different.
Expected length: 4
Actual length:   21
Strings differ at index 4.
Expected: 'boom'
But was:  'boomAt line:1 char:25'
at <ScriptBlock>, /home/vsts/work/1/s/test/powershell/Host/ConsoleHost.Tests.ps1: line 417
417:             $out.Split([Environment]::NewLine)[0] | Should -BeExactly "boom"
```
Write-Error Tests.ErrorRecord should not be truncated or have inserted newlines when redirected from another process
```powershell
Expected exactly 3, but got 1.
at <ScriptBlock>, /home/vsts/work/1/s/test/powershell/Modules/Microsoft.PowerShell.Utility/Write-Error.Tests.ps1: line 105
105:         $result.Count | Should -BeExactly 3
```

Comment (PoshChan) [inline src/System.Management.Automation/FormatAndOutput/DefaultFormatters/PowerShellCore_format_ps1xml.cs]: @SteveL-MSFT, your last commit had 2 failures in `PowerShell-CI-macos`
ConsoleHost unit tests.Pipe to/from powershell.errors are in text if error is redirected, encoded command, non-interactive, and outputformat specified
```powershell
Expected strings to be the same, but they were different.
Expected length: 4
Actual length:   21
Strings differ at index 4.
Expected: 'boom'
But was:  'boomAt line:1 char:25'
at <ScriptBlock>, /Users/runner/runners/2.164.6/work/1/s/test/powershell/Host/ConsoleHost.Tests.ps1: line 417
417:             $out.Split([Environment]::NewLine)[0] | Should -BeExactly "boom"
```
Write-Error Tests.ErrorRecord should not be truncated or have inserted newlines when redirected from another process
```powershell
Expected exactly 3, but got 1.
at <ScriptBlock>, /Users/runner/runners/2.164.6/work/1/s/test/powershell/Modules/Microsoft.PowerShell.Utility/Write-Error.Tests.ps1: line 105
105:         $result.Count | Should -BeExactly 3
```

Comment (PoshChan) [inline src/System.Management.Automation/FormatAndOutput/DefaultFormatters/PowerShellCore_format_ps1xml.cs]: @SteveL-MSFT, your last commit had 2 failures in `PowerShell-CI-windows`
ConsoleHost unit tests.Pipe to/from powershell.errors are in text if error is redirected, encoded command, non-interactive, and outputformat specified
```powershell
Expected strings to be the same, but they were different.
Expected length: 4
Actual length:   21
Strings differ at index 4.
Expected: 'boom'
But was:  'boomAt line:1 char:25'
at <ScriptBlock>, D:\a\1\s\test\powershell\Host\ConsoleHost.Tests.ps1: line 417
417:             $out.Split([Environment]::NewLine)[0] | Should -BeExactly "boom"
```
Write-Error Tests.ErrorRecord should not be truncated or have inserted newlines when redirected from another process
```powershell
Expected exactly 3, but got 1.
at <ScriptBlock>, D:\a\1\s\test\powershell\Modules\Microsoft.PowerShell.Utility\Write-Error.Tests.ps1: line 105
105:         $result.Count | Should -BeExactly 3
```

File: test/powershell/engine/Formatting/ErrorView.Tests.ps1 [hunk 2]
```diff
@@ -61,12 +67,26 @@ Describe 'Tests for $ErrorView' -Tag CI {
...
             $e | Out-String | Should -BeLike "*$testScriptPath*"
             $e | Out-String | Should -Not -BeLike '*pester*'
         }
+
+        It "Long lines should be rendered correctly" {
+            $testscript = @'
+                        $myerrors = [System.Collections.ArrayList]::new()
+                        Copy-Item (New-Guid) (New-Guid) -ErrorVariable +myerrors -ErrorAction SilentlyContinue
+                $error[0]
+'@
```
Comment (iSazonov) [inline test/powershell/engine/Formatting/ErrorView.Tests.ps1]: If these indentation sizes are important please add a protection comment.

File: test/powershell/engine/Formatting/ErrorView.Tests.ps1 [hunk 3]
```diff
@@ -61,12 +67,26 @@ Describe 'Tests for $ErrorView' -Tag CI {
...
+
+            Set-Content -Path $testScriptPath -Value $testScript
+            $e = & $testScriptPath
+            $e | Out-String | Should -BeLike "*${testScriptPath}:2*"
+            # validate line number is shown
+            $e | Out-String | Should -BeLike '* 2 *'
```
Comment (iSazonov) [inline test/powershell/engine/Formatting/ErrorView.Tests.ps1]: We do `$e | Out-String` twice - maybe use a variable?

File: test/powershell/engine/Formatting/ErrorView.Tests.ps1 (hunk 2 above)
Comment (SteveL-MSFT) [reply test/powershell/engine/Formatting/ErrorView.Tests.ps1]: Will add a comment

File: test/powershell/engine/Formatting/ErrorView.Tests.ps1 (hunk 3 above)
Comment (SteveL-MSFT) [reply test/powershell/engine/Formatting/ErrorView.Tests.ps1]: Good suggestion

**Comment by iSazonov** [comment]
We have many issues with these helper functions. Maybe put them in a temp module? This can simplify debugging and fixing.

**Comment by SteveL-MSFT** [comment]
@iSazonov which helpers are you referring to?  The tests don't really have any helpers?

**Comment by iSazonov** [comment]
@SteveL-MSFT I mean the formatting functions like Show-Error and Get-ConciseViewPositionMessage.

File: test/powershell/engine/Formatting/ErrorView.Tests.ps1 [hunk 4]
```diff
@@ -34,12 +41,11 @@ Describe 'Tests for $ErrorView' -Tag CI {
...
 
-            $testScriptPath = Join-Path -Path $TestDrive -ChildPath 'test.ps1'
             Set-Content -Path $testScriptPath -Value $testScript
-            $e = { & $testScriptPath } | Should -Throw -ErrorId 'UnexpectedToken' -PassThru
-            $e | Out-String | Should -BeLike "*${testScriptPath}:4*"
+            $e = { & $testScriptPath } | Should -Throw -ErrorId 'UnexpectedToken' -PassThru | Out-String
+            $e | Should -BeLike "*${testScriptPath}:4*"
             # validate line number is shown
-            $e | Out-String | Should -BeLike '* 4 *'
+            $e | Should -BeLike '* 4 *'
```
Comment (JamesWTruher) [inline test/powershell/engine/Formatting/ErrorView.Tests.ps1]: I'm not sure why there are 2 assertions here, it seems like one would be sufficient

**Review by JamesWTruher — APPROVED** [review]
All this code seems pretty fragile and relies on the format of the error message which could change at any time. There's not much we can do about other that to rethink the whole error presentation which is probably not something we should entertain at the moment.

File: test/powershell/engine/Formatting/ErrorView.Tests.ps1 (hunk 4 above)
Comment (SteveL-MSFT) [reply test/powershell/engine/Formatting/ErrorView.Tests.ps1]: They are similar but validate different parts of the rendering

**Comment by SteveL-MSFT** [comment]
@iSazonov In 7.1, we'll have appropriate APIs for coloring/VT100, we can consider some refactoring then vs making a big change now

**Review by TravisEz13 — APPROVED** [review]

**Pull request merged**